pkg/scheduler/core/generic_scheduler.go

```diff
@@ -274,7 +278,7 @@ func (g *genericScheduler) findNodesThatPassFilters(ctx context.Context, fwk fra
 		return nil, err
 	}
 
-	numNodesToFind := g.numFeasibleNodesToFind(int32(len(allNodes)))
+	numNodesToFind := g.numFeasibleNodesToFind(fwk, int32(len(allNodes)))
```

Comment:
If we followed @ahg-g's comment at https://github.com/kubernetes/kubernetes/pull/97263/files#r552730847 to remove the global percentage, `percentageOfNodesToScore` would be a profile-level field only, semantically its logic shouldn't be tightened to `g` (generic scheduler). I'd suggest to

- either expose `NodesNumToScore()` in framework/inteface.go, and hence here we just call `fwk.NodesNumToScore(int32(len(allNodes)))`.
- or (preferrable IMO), make the function stateless, i.e., `NodesNumToScore()` to accept two parameters: percentage and num of all nodes.